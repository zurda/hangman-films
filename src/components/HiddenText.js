import React from 'react'
import styled from '@emotion/styled'
import { colours } from '../style/shared'

const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 16px 32px;
`

const LetterContainer = styled.span`
  padding: 0 8px;
  display: inline-block;
  border-bottom: ${(props) =>
    props.letter === ' ' ? 'none' : `2px solid ${colours.maastrichtBlue}`};
  width: 20px;
  padding: 2px;
  text-align: center;
  margin-right: 4px;
  font-size: 20px;
  @media (min-width: 420px) {
    font-size: 22px;
  }
`

const HiddenText = ({ filmArr, guessedLetters }) => (
  <Container>
    {filmArr.map((letter, index) => (
      <LetterContainer key={index} letter={letter}>
        {guessedLetters.indexOf(letter) > -1 ? letter : '?'}
      </LetterContainer>
    ))}
  </Container>
)

export default HiddenText
