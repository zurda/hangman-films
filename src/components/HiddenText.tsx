import React from 'react'
import styled from '@emotion/styled'
import { colours } from '../style/shared'

const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 16px 32px;
  font-family: 'Roboto';
`

interface ILetterContainer {
  letter: string
}

const LetterContainer = styled.span<ILetterContainer>`
  padding: 0 8px;
  display: inline-block;
  border-bottom: ${(props) =>
    props.letter === ' ' ? 'none' : `2px solid ${colours.maastrichtBlue}`};
  width: 20px;
  padding: 2px;
  text-align: center;
  margin-right: 4px;
  font-size: 20px;
  @media (min-width: 420px) {
    font-size: 22px;
  }
`

interface IFilmArr {
  filmArr: string[]
  guessedLetters: string[]
}

const HiddenText = ({ filmArr, guessedLetters }: IFilmArr) => (
  <Container>
    {filmArr.map((letter, index) => (
      <LetterContainer key={index} letter={letter}>
        {guessedLetters.indexOf(letter) > -1 ? letter : '?'}
      </LetterContainer>
    ))}
  </Container>
)

export default HiddenText
