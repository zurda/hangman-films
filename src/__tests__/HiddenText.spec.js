import React from 'react'
import { create } from 'react-test-renderer'
import HiddenText from '../components/HiddenText'

const guessedLetters = [' ', 'a', 'b', 'c']
const filmArr = ['S', 'o', 'm', 'e', ' ', 'f', 'i', 'l', 'm']

describe('LettersTray component', () => {
  test('Matches the snapshot', () => {
    const hiddenText = create(
      <HiddenText guessedLetters={guessedLetters} filmArr={filmArr} />
    )
    expect(hiddenText.toJSON()).toMatchSnapshot()
  })
})
