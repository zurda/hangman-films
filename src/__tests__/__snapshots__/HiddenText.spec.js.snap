// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LettersTray component Matches the snapshot 1`] = `
<div
  className="css-87q7qv"
>
  <span
    className="css-h5d4jr"
  >
    ?
  </span>
  <span
    className="css-h5d4jr"
  >
    ?
  </span>
  <span
    className="css-h5d4jr"
  >
    ?
  </span>
  <span
    className="css-h5d4jr"
  >
    ?
  </span>
  <span
    className="css-1ep591g"
  >
     
  </span>
  <span
    className="css-h5d4jr"
  >
    ?
  </span>
  <span
    className="css-h5d4jr"
  >
    ?
  </span>
  <span
    className="css-h5d4jr"
  >
    ?
  </span>
  <span
    className="css-h5d4jr"
  >
    ?
  </span>
</div>
`;
